### PR TITLE
Remove the need to include a copy of the default app config to add a provider

### PR DESCRIPTION
**Pull request by Lidbetter**
This is an attempt at inserting the new providers into the providers array (in app config) rather than replace the whole thing.

This method of inserting providers into the config file should work (with some minor tweaking) for any laravel package which includes an install command.

General flow is as follows:
1) get the interpreted config array
2) check the provider needs to be added
3) using the interpreted array, get the last key of the providers array
4) get the file contents of the config file, and insert the new provider after the existing last key of the provider array

Currently it relies on finding the end of the array using short array syntax (but should be easy to support `array();`, if desired).

___
__Why?__

1) People update their app config files, installing spark should not undo that!
2) The spark installer does not need a version bump if the framework app config changes which providers are loaded by default

### CODE REVIEW

File: src/Console/InstallCommand.php [hunk 1]
```diff
@@ -57,4 +69,130 @@ private function findComposer()
...
+    }
+
+    /**
+     * The raw sourcecode string to inset into the app config file
+     *
+     * @return string
+     */
+    protected function sourcecodeToInsert()
+    {
+        $sparkProviderSource = PHP_EOL;
```
Comment (GrahamCampbell) [inline src/Console/InstallCommand.php]: Please don't use PHP_EOL

Comment (GrahamCampbell) [inline src/Console/InstallCommand.php]: You should use `\n`.

Comment (GrahamCampbell) [inline src/Console/InstallCommand.php]: Laravel follows PSR-2.

**Comment by taylorotwell** [comment]
Why not just have a check in the autoload file to see if we are running in a console instead of adding an entirely separate file?

**Comment by Lidbetter** [comment]
I don't follow, I didn't add any files

**Comment by GrahamCampbell** [comment]
I assume Taylor's comment was aimed at a different PR he just closed, lol.

**Comment by Lidbetter** [comment]
ok, I made the newline change.

If this was intentionally closed, I would appreciate a little feedback as to why.

I think a stub file is nicer so I will change that if this is reopened.

**Pull request closed without merging**